### PR TITLE
Upgrade to go1.9.2

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Use go1.9.2, containing a number of bug fixes: https://github.com/golang/go/issues?q=milestone%3AGo1.9.2

**Release note**:
```release-note
Upgrade to go1.9.2
```

### CODE REVIEW

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -1,8 +1,8 @@
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "441e560e947d8011f064bd7348d86940d6b6131ae7d7c4425a538e8d9f884274",
-    strip_prefix = "rules_go-c72631a220406c4fae276861ee286aaec82c5af2",
-    urls = ["https://github.com/bazelbuild/rules_go/archive/c72631a220406c4fae276861ee286aaec82c5af2.tar.gz"],
+    sha256 = "e09aa1a3318a292c24b3c97584f2fc3ceb7636ec786e580d1adba1efce320bb7",
+    strip_prefix = "rules_go-b358831ed503659656daa35a361094a1eee5aa60",
+    urls = ["https://github.com/bazelbuild/rules_go/archive/b358831ed503659656daa35a361094a1eee5aa60.tar.gz"],
```
Comment (ixdy) [inline build/root/WORKSPACE]: Looks like this broke `go_genrule` again, unsurprisingly. I will take a look (unless you want to dive into skylark yourself).

**Review by luxas — COMMENTED** [review]
/lgtm

(ignorant to the bazel fix there, but please investigate why it broke indeed :smile:)

File: build/root/WORKSPACE [hunk 2]
```diff
@@ -1,8 +1,8 @@
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "441e560e947d8011f064bd7348d86940d6b6131ae7d7c4425a538e8d9f884274",
-    strip_prefix = "rules_go-c72631a220406c4fae276861ee286aaec82c5af2",
-    urls = ["https://github.com/bazelbuild/rules_go/archive/c72631a220406c4fae276861ee286aaec82c5af2.tar.gz"],
+    sha256 = "8e51b3d7902a0ed69e84ff7c30c2141d74970b8caa8b44bc2f6eded76f0abdef",
+    strip_prefix = "rules_go-3fd320621566c44aad0f7df5d0f5858efd2bbcf6",
```
Comment (ixdy) [inline build/root/WORKSPACE]: can you please revert these changes and instead cherry-pick in https://github.com/ixdy/kubernetes/commit/3854c3a6f9bfe4b502fa6910ad31751e3fdef88d?

Comment (cblecker) [reply build/root/WORKSPACE]: Done!

File: hack/update-bazel.sh [hunk 3]
```diff
@@ -32,7 +32,7 @@ kube::util::go_install_from_commit \
     ae4e9a3906ace4ba657b7a09242610c6266e832c
 kube::util::go_install_from_commit \
     github.com/bazelbuild/rules_go/go/tools/gazelle/gazelle \
-    c72631a220406c4fae276861ee286aaec82c5af2
+    3fd320621566c44aad0f7df5d0f5858efd2bbcf6
```
Comment (ixdy) [inline hack/update-bazel.sh]: revert this too?

**Review by ixdy — COMMENTED** [review]
I'll go build/push the crossbuild image. will update here when done.

File: hack/update-bazel.sh (hunk 3 above)
Comment (cblecker) [reply hack/update-bazel.sh]: crud. yup, doing that now

**Comment by cblecker** [comment]
@ixdy done

**Comment by dims** [comment]
`gcr.io/google_containers/kube-cross:v1.9.2-1` needs to be pushed? the cross build seems to not find it

**Comment by ixdy** [comment]
yep, building it now.

**Comment by ixdy** [comment]
and pushed.

/lgtm
/retest

**Comment by ixdy** [comment]
(we might want to update the minimum go version in `hack/lib/golang.sh` from 1.9.1 to 1.9.2, though that's not super critical IMO.)

**Comment by dims** [comment]
LGTM 👍
